packages/nestjs-custom-decorators/src/decode-jwt.decorator.ts

```diff
@@ -0,0 +1,17 @@
+import { createParamDecorator, ExecutionContext } from '@nestjs/common';
+import * as jwtDecode from 'jwt-decode';
+// tslint:disable-next-line:variable-name
+export const DecodeJWT: () => ParameterDecorator = createParamDecorator(
+  async (_data: unknown, ctx: ExecutionContext) => {
+    const request: { accessTokenJWT: string } = ctx.switchToHttp().getRequest();
+    const jwt: string = request.accessTokenJWT;
+    let decodedJwt: object;
+    try {
+      decodedJwt = jwtDecode(jwt);
+    } catch (err) {
+      throw new Error('UNAUTHORIZED');
```

Comment:
Import this class from @nest/common lib

```suggestion
      throw new UnauthorizedException('UNAUTHORIZED');
```